### PR TITLE
RSDK-6987: sensor-controlled base clean up of velocities, move straight, and spin files

### PR DESCRIPTION
**Pull request by martha-johnston**
mostly minor changes to make all the sensor-controlled base files more consistent with each other. need to do more testing to make sure the polling and cancelling logic still works as expected

### CODE REVIEW

File: components/base/sensorcontrolled/velocities.go [hunk 1]
```diff
@@ -3,159 +3,37 @@ package sensorcontrolled
...
-	// set the spin loop to false, so we do not skip the call to SetState in the control loop
-	// this will also stop any active Spin calls
-	sb.setPolling(false)
-
-	if len(sb.controlLoopConfig.Blocks) != 0 {
-		// start a sensor context for the sensor loop based on the longstanding base
-		// creator context, and add a timeout for the context
-		timeOut := 10 * time.Second
-		var sensorCtx context.Context
-		sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
```
Comment (martha-johnston) [inline components/base/sensorcontrolled/velocities.go:91]: this is the main think I'm not sure about removing. there seems to be no issues with the calls interrupting each other from the RC card, but I need to spam it a little and also test from a client script

**Review by JohnN193 — COMMENTED** [review]
Looks good, only question I have is if we want to move `State()` and `SetState()` into `sensorcontrolled.go`. maybe `sign()` as well?

**Comment by martha-johnston** [comment]
> Looks good, only question I have is if we want to move `State()` and `SetState()` into `sensorcontrolled.go`. maybe `sign()` as well?

yes super good point. will do

**Comment by martha-johnston** [comment]
well actually now there's like nothing in the velocities file since it's so simple. so maybe move all the control loop stuff back to velocities.go? what do you think @JohnN193

**Review by penguinland — COMMENTED** [review]
It sounds like John has a better sense of what this code is doing than I do. I can continue trying to review if you want, but he'll likely give better feedback.

File: components/base/sensorcontrolled/movestraight.go [hunk 2]
```diff
@@ -22,35 +22,37 @@ const (
...
 	}
 
+	straightTimeEst := time.Duration(int(time.Second) * int(math.Abs(float64(distanceMm)/mmPerSec)))
+	startTime := time.Now()
+	timeOut := 5 * straightTimeEst
+	if timeOut < 10*time.Second {
+		timeOut = 10 * time.Second
+	}
+
+	initialHeading, err := sb.headingFunc(ctx)
```
Comment (JohnN193) [inline components/base/sensorcontrolled/movestraight.go:51]:
```suggestion
	// grab the initial heading for MoveStraight to clamp to. Will return 0 if no supporting sensors were configured.
	initialHeading, err := sb.headingFunc(ctx)
```

File: components/base/sensorcontrolled/movestraight.go [hunk 3]
```diff
@@ -22,35 +22,37 @@ const (
 func (sb *sensorBase) MoveStraight(
```
Comment (JohnN193) [inline components/base/sensorcontrolled/movestraight.go:22]: [q]should we add a comment for each function describing how it uses controls?

**Review by JohnN193 — APPROVED** [review]
LGTM, we could maybe add some comments but I am happy with us removing all of the older functions

File: components/base/sensorcontrolled/movestraight.go (hunk 3 above)
Comment (martha-johnston) [reply components/base/sensorcontrolled/movestraight.go:22]: yeah good idea

File: components/base/sensorcontrolled/movestraight.go [hunk 4]
```diff
@@ -19,38 +19,45 @@ const (
 	headingGain           = 1.
 )
 
+// MoveStraight commands a base to move forward for the desired distanceMm at the given mmPerSec.
+// When controls are enabled, MoveStraight calculates the required velocity to reach mmPerSec
+// and the distanceMm goal. It then polls the provided velocity movement sensor and corrects any
+// error between this calculated velocity and the actual velocity using a PID control loop.
```
Comment (martha-johnston) [inline components/base/sensorcontrolled/movestraight.go:25]: @JohnN193 is this true? I see nothing about the velocity movement sensor but I assume it has to use that? might need to add a check for it

Comment (JohnN193) [reply components/base/sensorcontrolled/movestraight.go:25]: Yeah, it uses the velocity sensor via the pid loop.

its also tracking position to stop moving when it reaches the goal, and attempts to keep the heading of the rover fixed if an appropriate sensor is provided

Comment (JohnN193) [reply components/base/sensorcontrolled/movestraight.go:25]: The control loop isn’t configured unless velocity exists already

Comment (martha-johnston) [reply components/base/sensorcontrolled/movestraight.go:25]: right but we are checking in line 33 if position or control loop exists. what if you don't have a velocity movement sensor, so  no control loop, but you do have a position reporting sensor? we will run into issues where we try to create a control loop when we don't have a control loop config

Comment (martha-johnston) [reply components/base/sensorcontrolled/movestraight.go:25]: okay I understand now nvm

Comment (martha-johnston) [reply components/base/sensorcontrolled/movestraight.go:25]: I'll just add a comment

File: components/base/sensorcontrolled/movestraight.go [hunk 5]
```diff
@@ -19,38 +19,51 @@ const (
...
-	straightTimeEst := time.Duration(int(time.Second) * int(math.Abs(float64(distanceMm)/mmPerSec)))
-	startTime := time.Now()
-	timeOut := 5 * straightTimeEst
-	if timeOut < 10*time.Second {
-		timeOut = 10 * time.Second
-	}
 
+	// If a position movement sensor or controls are not configured, we cannot use this MoveStraight method.
+	// Instead we need to use the MoveStraight method of the base that the sensorcontrolled base wraps.
+	// If there is no valid velocity sensor, there won't be a controlLoopConfig.
```
[the remark this answers is not in the document]
Comment (martha-johnston) [reply components/base/sensorcontrolled/movestraight.go:38]: @JohnN193 before I merge does this comment make sense?

[the remark this answers is not in the document]
Comment (JohnN193) [reply components/base/sensorcontrolled/movestraight.go:38]: yeah this comment makes sense!

**Pull request merged**